src/cargo/util/toml/mod.rs

```diff
@@ -2839,6 +2921,79 @@ impl TomlManifest {
     }
 }
 
+fn parse_unstable_lints<T: Deserialize<'static>>(
+    lints: Option<toml::Value>,
+    features: &Features,
+    config: &Config,
+) -> CargoResult<Option<T>> {
+    let Some(lints) = lints else { return Ok(None); };
+
+    if !features.is_enabled(Feature::lints()) {
+        warn_for_feature("lints", config);
+        return Ok(None);
+    }
+
+    lints.try_into().map(Some).map_err(|err| err.into())
+}
+
+fn warn_for_feature(name: &str, config: &Config) {
```

Comment:
> Should this be a method of Features struct, like Features::require?

> I broke out the warn_for_feature as an experiment towards us systemitizing this stabilization approach which we also used with https://github.com/rust-lang/cargo/pull/9732. This works well when we can ignore the new information which isn't too often but sometimes happens.

Figured it'd be better to understand this usage of the warning before stabilizing

> Will the user get a warning from each dependency having lints table?

Yes, and I've switched it to use the existing warning system for manifests.  This will hide it for non-workspace members and show the path for workspace members.

Note that we strip the `[lints]` table on stable so it won't get published